File: py/pyis-python/tests/test_ngram_featurizer/test_ngram_featurizer.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT license.

import os
import unittest
from pyis.python import ops

class TestNGramFeaturizer(unittest.TestCase):
    def test_add(self):
        featurizer = ops.NGramFeaturizer(2, False)
        featurizer.fit(['hello', 'world'])
        features = featurizer.transform(['hello', 'world'])
        for i in features:
            print(i.to_tuple())

    def test_load(self):
        ngram_file = os.path.join(os.path.dirname(__file__), 'data', 'ngram.vocab.txt')
        featurizer = ops.NGramFeaturizer(1, False)
        featurizer.load_ngram(ngram_file)
        features = featurizer.transform(['hello', 'world'])
        for i in features:
            print(i.to_tuple())

if __name__ == '__main__':
    unittest.main()

File: tests/test_ngram_featurizer/test_ngram_featurizer.cpp
#include "gtest/gtest.h"
#include "pyis/ops/text/ngram_featurizer.h"

TEST(TestNGramFeaturizer, TestLoad) {
    std::string ngram_file = "tests/test_ngram_featurizer/data/ngram.txt";
    pyis::ops::NGramFeaturizer featurizer(1, false);
    featurizer.LoadNGram(ngram_file);

    auto features = featurizer.Transform({"hello", "world"});
    ASSERT_EQ(features.size(), 2);
    ASSERT_EQ(features[0].id(), 0);
    ASSERT_EQ(features[1].id(), 1);
}